### ISSUE TITLE
Misleading error on multiple labels which at least partially match a query

### ISSUE THREAD

**Issue by NobbZ** [keathley/wallaby#168]
## Environment

The phoenix project available at https://github.com/NobbZ/wallaby_bad_error_when_multiple_lables_match_a_query has a single page with a single form, containing two fields with a label each (see `lib/wallaby_bug/web/templates/page/index.html.eex`).

## Bug description

There are two tests defined in `test/acceptance/wallaby_example_test.exs`, from which the following fails:

```exs
test "it can't fill a form", %{session: session} do
  session
  |> visit("/")
  |> fill_in(Query.text_field("field"), with: "text")
  |> fill_in(Query.text_field("other field"), with: "more text")
  |> text
end
```

The message it fails with, calims that the `for` attribute points to something nonexisting. As a solution it suggests to add `id="field1"` to the corresponding field, which is already there.

```plain
  1) test it can't fill a form (WallabyBug.Web.WallabyExample)
     test/acceptance/wallaby_example_test.exs:4
     ** (Wallaby.QueryError) The text 'field' matched a label but the label's 'for' attribute
     doesn't match the id of any element.

     Make sure that id on your element is `id="field1"`.

     stacktrace:
       (wallaby) lib/wallaby/browser.ex:785: Wallaby.Browser.find/2
       (wallaby) lib/wallaby/browser.ex:750: Wallaby.Browser.find/3
       test/acceptance/wallaby_example_test.exs:7: (test)
```

as a reference there is a test in the same file which uses `Query.css(id)`-syntax, which passes.

## Expected behaviour

There are two mutually exclusive different possible expectations here:

1. Choose the first label, as it is the closest match and continue processing
2. Provide proper error message which tells that the query matches on multiple items and explain ways to narrow it down.

I'd prefer 2.

**Comment by keathley** [comment]
@NobbZ Is that the entire html? I'm assuming that there is a Password field on the page somewhere?

**Comment by NobbZ** [comment]
Well there is, and the message was copied from the wrong window from a run that already used css selection for email. The correct message that was shown to me of course referenced the word Email.

**Comment by PragTob** [comment]
@NobbZ if you could share the HTML with us so we can set up a test case to investigate and fix that'd be great :)

**Comment by NobbZ** [comment]
I will construct some minimal project which shows the problem during the next 2 days.

**Comment by PragTob** [comment]
Thanks a lot @NobbZ - that's a great help! :green_heart: :tada:

**Comment by NobbZ** [comment]
I created a repository at https://github.com/NobbZ/wallaby_bad_error_when_multiple_lables_match_a_query, I will update my initial description of the bug accordingly.

**Comment by PragTob** [comment]
Thanks a lot for putting in the work to reproduce @NobbZ ! This is an enormous help!

![thanks](https://media.giphy.com/media/3oz8xIsloV7zOmt81G/giphy.gif)

**Comment by keathley** [comment]
@NobbZ I just tested this with Wallaby 0.17 and its returning the correct error now. Thanks for that repo! It made it very easy to test with.

**Comment by keathley** [comment]
For the sake of maintaining history this was fixed as part of in #235.
